### ISSUE TITLE
CodeNamespace.Members does not include records

### ISSUE THREAD

**Issue by PhilPJL** [dotnet/roslyn#62523]
Not sure where or how to report this but was redirected here from the developer community forum. 
My original post is [https://developercommunity.visualstudio.com/t/CodeNamespaceMembers-does-not-include-r/10035834](https://developercommunity.visualstudio.com/t/CodeNamespaceMembers-does-not-include-r/10035834)

The issue boils down to EnvDTE does not currently support the record keyword.

**Comment by CyrusNajmabadi** [comment]
Yes, this is true. Codemodel is an extremely problematic legacy api. It's slow, mutable, and only usable from the ui thread. It is also not receiving updates. It has been superseded by the Roslyn syntactic and semantic apis. Those apis are much faster, immutable, and can be used safely off of the ui thread. 

Thanks!

**Comment by PhilPJL** [comment]
_It is also not receiving updates._

Point me at the MS statement that says that.  As I understand it Entity Framework Core 7 is planning to use T4 templates so I'm assuming they're not deprecated.

**Comment by CyrusNajmabadi** [comment]
> Point me at the MS statement that says that.

^ https://github.com/dotnet/roslyn/issues/62523#issuecomment-1179559126

We haven't updated code model, except for trivial issues, in more than a decade. It and codedom have been replaced with the Roslyn apis for getting full syntactic and semantic information about c# code.  It's one of the reasons we made Roslyn in the first place :-)

The codemodel design was fundamentally too limiting and unmaintainable. It had low fidelity, and a bunch of other issues due to it's COM roots. Roslyn *is* the modern replacement for it and has 100% fidelity with the language (and all it's updates) since the start, without any of those limitations.
